### PR TITLE
Added CSP headers for Aardvaark, as per BTS-196.

### PR DESCRIPTION
**Pull request by adityamukho**
### Scope & Purpose

This PR adds content security policy (CSP) headers to the Aardvaark app, to make it more secure against certain attacks. The headers added were as per the recommendations in [BTS-196](https://arangodb.atlassian.net/browse/BTS-196).

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.10: https://github.com/arangodb/arangodb/pull/16637

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/BTS-196
- [ ] Design document:

### CODE REVIEW

File: arangod/GeneralServer/CommTask.cpp [hunk 1]
```diff
@@ -375,6 +375,16 @@ void CommTask::finishExecution(GeneralResponse& res,
     // use "IfNotSet" to not overwrite an existing response header
     res.setHeaderNCIfNotSet(StaticStrings::XContentTypeOptions,
                             StaticStrings::NoSniff);
+
```
Comment (dothebart) [inline arangod/GeneralServer/CommTask.cpp:378]: do we care in case of VST? so, could it only be added for HTTP[1.1/2]?

**Review by dothebart — APPROVED** [review]

File: lib/Basics/StaticStrings.cpp [hunk 2]
```diff
@@ -186,7 +186,10 @@ std::string const StaticStrings::AsyncId("x-arango-async-id");
 std::string const StaticStrings::Authorization("authorization");
 std::string const StaticStrings::BatchContentType(
     "application/x-arango-batchpart");
-std::string const StaticStrings::CacheControl("cache-control");
+std::string const StaticStrings::CacheControl("Cache-Control");
+std::string const StaticStrings::CacheControlValue(
+    "no-cache, no-store, must-revalidate, pre-check=0, post-check=0, "
+    "max-age=0, s-maxage=0");
```
Comment (mpoeter) [inline lib/Basics/StaticStrings.cpp:192]: I would not define a static string for the value. It is only used once, and at the usage site it is not clear what the value actually is.

File: lib/Basics/StaticStrings.cpp [hunk 3]
```diff
@@ -229,6 +232,17 @@ std::string const StaticStrings::XContentTypeOptions("x-content-type-options");
...
+    "Content-Security-Policy");
+std::string const StaticStrings::ContentSecurityPolicyValue(
+    "frame-ancestors 'self'; form-action 'self';");
+std::string const StaticStrings::Pragma("Pragma");
+std::string const StaticStrings::PragmaValue("no-cache");
+std::string const StaticStrings::Expires("Expires");
+std::string const StaticStrings::ExpiresValue("0");
+std::string const StaticStrings::HSTS("Strict-Transport-Security");
+std::string const StaticStrings::HSTSValue(
+    "max-age=31536000 ; includeSubDomains");
```
Comment (mpoeter) [inline lib/Basics/StaticStrings.cpp:245]: Same as above.

File: lib/Basics/StaticStrings.cpp (hunk 2 above)
Comment (adityamukho) [reply lib/Basics/StaticStrings.cpp]: Fixed in 9cae2f96bcc

**Review by hkernbach — COMMENTED** [review]
LGTM. But first the discussion in https://github.com/arangodb/arangodb/pull/16506/files#r914520126 must be solved (Whether we need that change in HTTP only or if this should also be VST relevant. In terms of aardvark support - we do not have any VST communication there)

File: arangod/GeneralServer/CommTask.cpp (hunk 1 above)
Comment (adityamukho) [reply arangod/GeneralServer/CommTask.cpp:378]:
```
curl --http2 -v -i http://localhost:8529/
*   Trying 127.0.0.1:8529...
* Connected to localhost (127.0.0.1) port 8529 (#0)
> GET / HTTP/1.1
> Host: localhost:8529
> User-Agent: curl/7.84.0
> Accept: */*
> Connection: Upgrade, HTTP2-Settings
> Upgrade: h2c
> HTTP2-Settings: AAMAAABkAAQCAAAAAAIAAAAA
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 101 Switching Protocols
HTTP/1.1 101 Switching Protocols
< Connection: Upgrade
Connection: Upgrade
< Upgrade: h2c
Upgrade: h2c
* Received 101
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0

* Connection state changed (MAX_CONCURRENT_STREAMS == 32)!
* HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)
* Connection #0 to host localhost left intact
curl: (92) HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)
```

It seems HTTP/2 support is not fully functional. Will have to wait until it is, before testing headers for this protocol.

**Comment by adityamukho** [comment]
> LGTM. But first the discussion in https://github.com/arangodb/arangodb/pull/16506/files#r914520126 must be solved (Whether we need that change in HTTP only or if this should also be VST relevant. In terms of aardvark support - we do not have any VST communication there)

The headers are already only being set for HTTP, and not for VST.

**Review by mpoeter — APPROVED** [review]

**Review by hkernbach — APPROVED** [review]

**Comment by KVS85** [comment]
@adityamukho please address failures in tests.

File: arangod/GeneralServer/CommTask.cpp (hunk 1 above)
Comment (adityamukho) [reply arangod/GeneralServer/CommTask.cpp:378]: Correction: The HTTP/2 error was being caused by camel-cased headers. This is now fixed with all headers being sent in lowercase. To answer the original question, `Endpoint::TransportType::HTTP` is the type for HTTP/2 requests also.

**Comment by adityamukho** [comment]
Main matrix PR: https://jenkins01.arangodb.biz:3456/view/PR/job/arangodb-matrix-pr/22774/
Two subtasks failed due to unrelated errors: 

1. https://jenkins01.arangodb.biz:3456/job/arangodb-matrix-pr-linux.x86-64/23172/
2. https://jenkins01.arangodb.biz:3456/job/arangodb-matrix-pr-linux.aarch64/1934/

These were re-run successfully with the following job links respectively:

1. https://jenkins01.arangodb.biz:3456/job/arangodb-matrix-pr-linux.x86-64/23178/
2. https://jenkins01.arangodb.biz:3456/job/arangodb-matrix-pr-linux.aarch64/1938/

**Review by dothebart — APPROVED** [review]

**Review by hkernbach — APPROVED** [review]

**Pull request merged**